src/Pos/Delegation/Class.hs

```diff
@@ -56,9 +54,6 @@ data DelegationWrap = DelegationWrap
       -- a good approximation.
     , _dwEpochId           :: EpochIndex
       -- ^ Epoch index 'DelegationWrap' is correct in relation to.
-    , _dwThisEpochPosted   :: HashSet PublicKey
```

Comment:
Why did you remove it from the context? It doesn't take much space and it's faster to take this data from memory than from DB.